lua/gitlab/actions/create_mr.lua

```diff
@@ -288,19 +278,27 @@ M.create_mr = function()
 end
 
 M.create_layout = function()
-  local title_popup = Popup(title_popup_settings)
+  local title_popup = Popup(u.create_box_popup_state("Title", false))
   M.title_bufnr = title_popup.bufnr
-  local description_popup = Popup(description_popup_settings)
+  local description_popup = Popup(u.create_box_popup_state("Description", true))
   M.description_bufnr = description_popup.bufnr
-  local target_branch_popup = Popup(target_popup_settings)
+  local target_branch_popup = Popup(u.create_box_popup_state("Target branch", false))
   M.target_bufnr = target_branch_popup.bufnr
+  local delete_branch_popup = Popup(u.create_box_popup_state("Delete Source Branch", false))
+  M.delete_branch_bufnr = delete_branch_popup.bufnr
+  local squash_popup = Popup(u.create_box_popup_state("Squash Commits", false))
+  M.squash_bufnr = squash_popup.bufnr
 
   local internal_layout
   internal_layout = Layout.Box({
     Layout.Box({
       Layout.Box(title_popup, { grow = 1 }),
       Layout.Box(target_branch_popup, { grow = 1 }),
     }, { size = 3 }),
+    Layout.Box({
+      Layout.Box(delete_branch_popup, { grow = 1 }),
+      Layout.Box(squash_popup, { grow = 1 }),
+    }, { size = 3 }),
```

Comment:
I also experimented with a different layout where all the small popups (title, branch, delete, squash) would be on a single line with different popup widths, but I thought this might not work well on a very narrow (or split) screen.

Reply:
I'm inclined to actually put the metadata fields _below_ the description, like they are on Gitlab. If you shorten the titles a bit they can fit easily, even on smaller screen sizes: 

![Screenshot 2024-04-04 at 7 17 03 PM](https://github.com/harrisoncramer/gitlab.nvim/assets/32515581/b26a07a4-3e43-4051-a995-6f988de55649)

This is achieved like so: 

```lua
  internal_layout = Layout.Box({
    Layout.Box({
      Layout.Box(title_popup, { grow = 1 }),
    }, { size = 3 }),
    Layout.Box(description_popup, { grow = 1 }),
    Layout.Box({
      Layout.Box(target_branch_popup, { grow = 1 }),
      Layout.Box(delete_branch_popup, { grow = 1 }),
      Layout.Box(squash_popup, { grow = 1 }),
    }, { size = 3 }),
  }, { dir = "col" })
```

Reply:
Oh, nice! I like the metadata below the description. Since "Delete source branch" and "Squash commits" don't need to grow, I'd put them to the left with a fixed width and let the "Target branch" take up the rest of the window, like this:
![image](https://github.com/harrisoncramer/gitlab.nvim/assets/17056143/087675a5-3676-4d29-ab91-72948f01563a)
And truncate the titles when the window length is below some limit, like this:
![image](https://github.com/harrisoncramer/gitlab.nvim/assets/17056143/09144dba-6a30-4b49-a65e-2a1d95ed0702)

This is fairly easy to do:
```lua
  local delete_title = vim.fn.winwidth(0) > 110 and "Delete source branch" or "Delete source"
  local delete_branch_popup = Popup(u.create_box_popup_state(delete_title, false))
  M.delete_branch_bufnr = delete_branch_popup.bufnr
  local squash_title = vim.fn.winwidth(0) > 110 and "Squash commits" or "Squash"
  local squash_popup = Popup(u.create_box_popup_state(squash_title, false))
  M.squash_bufnr = squash_popup.bufnr

  local internal_layout
  internal_layout = Layout.Box({
    Layout.Box({
      Layout.Box(title_popup, { grow = 1 }),
    }, { size = 3 }),
    Layout.Box(description_popup, { grow = 1 }),
    Layout.Box({
      Layout.Box(delete_branch_popup, { size = {width = #delete_title + 4} }),
      Layout.Box(squash_popup, { size = {width = #squash_title + 4} }),
      Layout.Box(target_branch_popup, { grow = 1 }),
    }, { size = 3 }),
  }, { dir = "col" })
```

Reply:
Looks good to me! Can you push up your changes?